src/plugins/simulator/xplane/simulatorxplane.cpp

```diff
@@ -1076,7 +1088,13 @@ namespace BlackSimPlugin::XPlane
             const CInterpolationResult result = xplaneAircraft.getInterpolation(currentTimestamp, setup, aircraftNumber++);
             if (result.getInterpolationStatus().hasValidSituation())
             {
-                const CAircraftSituation interpolatedSituation(result);
+                CAircraftSituation interpolatedSituation(result);
+
+                // adjust altitude to compensate for XP12 temperature effect
+                const CLength relativeAltitude = interpolatedSituation.geodeticHeight() - getOwnAircraftPosition().geodeticHeight();
+                const double altitudeDeltaWeight = qBound(3000.0, relativeAltitude.abs().value(CLengthUnit::ft()), 6000.0) / 3000 - 1;
```

Comment:
Same as above:
This weight should be 1 for the relativeAltitude between 0-3000ft (fully apply offset) and linearly fade out till 6000ft.
If I am not mistaken it seems that the weight is 1 for 6.000ft only and decreases to reach 0 at 3000 ft.

Reply:
+1